### PR TITLE
sum_edge_to_cells as dawn integration test

### PR DESCRIPTION
**Pull request by havogt**
## Technical Description
Integrates the unstructured sum_edge_to_cells
as a dawn integration test (IIR->serialize->deserialize)

Fixes bugs in ReductionOverNeighborExpr

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: dawn/test/integration-test/GenerateInMemoryStencils.cpp [hunk 1]
```diff
@@ -323,3 +324,27 @@ createLapStencilIIRInMemory(OptimizerContext& optimizer) {
...
+      b.stencil(b.multistage(
+          LoopOrderKind::Parallel,
+          b.stage(LocType::Edges, b.vregion(dawn::sir::Interval::Start, dawn::sir::Interval::End,
+                                            b.stmt(b.assignExpr(b.at(in_f), b.lit(10))))),
+          b.stage(b.vregion(dawn::sir::Interval::Start, dawn::sir::Interval::End,
+                            b.stmt(b.assignExpr(
+                                b.at(out_f), b.reduceOverNeighborExpr(
+                                                 Op::plus, b.at(in_f, HOffsetType::withOffset, 0),
+                                                 b.lit(0.), LocType::Cells, LocType::Edges))))))));
+  return (*stencil_instantiation.begin()).second;
```
Comment (mroethlin) [inline dawn/test/integration-test/GenerateInMemoryStencils.cpp]: maybe `return stencil_instantiation.at("generated")`is clearer

Comment (havogt) [reply dawn/test/integration-test/GenerateInMemoryStencils.cpp]: done.

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**